### PR TITLE
Filter on user fields

### PR DESCRIPTION
**Pull request by kellyel**
This PR allows users to filter on fields other than scores in the score report, as well as sorting.

### CODE REVIEW

File: src/helpers/query/assignments.js [hunk 1]
```diff
@@ -81,20 +85,33 @@ export const getAssignmentsRequestBody = ({
...
+      // const userFilters = (filters.map(filter => {
+      //   console.log('filter ->', filter)
+      //   return {
+      //     fieldFilter: {
+      //       field: { fieldPath: filter.field },
+      //       op: "EQUAL", // Need more logic here later for different constraints
+      //       value: { stringValue: filter.value }
+      //     }
+      //   }
+      // }))
```
Comment (richford) [inline src/helpers/query/assignments.js:111]:
```suggestion
```

File: src/helpers/query/assignments.js [hunk 2]
```diff
@@ -609,17 +647,42 @@ export const assignmentPageFetcher = async (
...
+    //     aggregationQuery: false,
+    //     pageLimit: pageLimit.value,
+    //     page: page.value,
+    //     paginate: paginate,
+    //     filter: filters,
+    //   });
+    //   console.log('Request Body', requestBody);
+    //   return adminAxiosInstance.post(':runQuery', requestBody).then(async ({ data }) => {
+    //     console.log('Found Data:', data);
+    //   });
```
Comment (richford) [inline src/helpers/query/assignments.js:665]:
```suggestion
```

**Review by richford — CHANGES_REQUESTED** [review]
This looks great @kellyel! Thanks for all of your hard work. I have two minor requests and one major one.

- Sorting by grade results in no data table to be displayed.
- I made two suggestions for removing commented out code.

**Comment by kellyel** [comment]
Feedback addressed:
- Change wording of filters in dropdown
- Allow filters to be applied to scores from any view (color, percentile, standard, raw)
- Prevent user from filtering on blacklisted columns (letter)
- Visually remove option to filter on blacklisted columns. I am as of right now unable to remove the button itself, however I have disabled sorting on that column (no action happens), and I was able to remove the icon for those columns. the only indication that there is a button there is the hover event. I am not 100% happy with this, and will return to the problem later.

**Review by richford — APPROVED** [review]

**Pull request merged**